scala-generator/src/main/scala/models/http4s/ScalaClientMethodGenerator.scala

```diff
@@ -107,7 +107,13 @@ class ScalaClientMethodGenerator (
           if (response.isUnit) {
             s"case r => ${http4sConfig.wrappedAsyncType("Sync").getOrElse(http4sConfig.asyncType)}.${http4sConfig.asyncFailure}(new errors.${response.errorClassName}(r.${config.responseStatusMethod}))"
           } else {
-            s"case r => ${http4sConfig.wrappedAsyncType("Sync").getOrElse(http4sConfig.asyncType)}.${http4sConfig.asyncFailure}(new errors.${response.errorClassName}(r))"
+            config match {
+              case _@(_: ScalaClientMethodConfigs.Http4s017 | _: ScalaClientMethodConfigs.Http4s015) =>
```

Comment:
this looks it can be simplified to

`case _: ScalaClientMethodConfigs.Http4s017 | _: ScalaClientMethodConfigs.Http4s015 =>`